### DISCUSSION TITLE
after response for get refresh token

### DISCUSSION THREAD

**Discussion by dodiwknd** [sindresorhus/ky#381] [Q&A] [answered]
I may be a bit confused about how to explain it, I hope those who read this understand.

in my case I'm trying to fetch a refresh token on the `afterResponse` hook, and if the API doesn't return the token I'll reject the request. but here I didn't manage to do that, with an example like this.
```typescript
const authAfterResponseHooks: AfterResponseHook = async (req, _, res) => {
  if (res.status === 401) {
      const token = await base
        .post('integration/customer/token')
        .json<string>();

      req.headers.set('Authorization', 'Bearer ' + token);

      if (typeof token !== 'string') {
        return Promise.reject(new Response('Errors token not get', { status: 400 }));
      }

      await Storage.setItem('token', token);
      return base(req);
  }
};
```
note: `base` is still executed, but the one under check is not executed

maybe someone can shed some light on this matter. thanks 🙏🏼

**Comment by essana3** [comment]
Shouldn't you use the `beforeError` hook ?

**Comment by sindresorhus** [comment, marked as answer]
We have an example of this in the docs: https://github.com/sindresorhus/ky#hooksafterresponse
